### PR TITLE
Align naming in Content Launcher cluster XML with the spec.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
REVIEW NOTE: Only the first commit contains manual changes. The rest is code-generated.

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml [hunk 1]
```diff
@@ -30,83 +30,84 @@ limitations under the License.
 
     <command source="client" code="0x00" name="LaunchContent" response="LaunchResponse" optional="true">
       <description>Upon receipt, this SHALL launch the specified content with optional search criteria.</description>
-      <arg name="search" type="ContentSearch"/>
-      <arg name="autoPlay" type="BOOLEAN"/>
-      <arg name="data" type="CHAR_STRING" optional="true"/>
+      <arg name="Search" type="ContentSearchStruct"/>
+      <arg name="AutoPlay" type="BOOLEAN"/>
+      <arg name="Data" type="CHAR_STRING" optional="true"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:35]: spec seems to say octet string

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml [hunk 2]
```diff
@@ -30,83 +30,84 @@ limitations under the License.
...
+      <arg name="DisplayString" type="CHAR_STRING" optional="true"/>
+      <arg name="BrandingInformation" type="BrandingInformationStruct" optional="true"/>
     </command>
 
     <command source="server" code="0x02" name="LaunchResponse" optional="true" disableDefaultResponse="true">
       <description>This command SHALL be generated in response to LaunchContent command.</description>
-      <arg name="status" type="ContentLaunchStatusEnum"/>
-      <arg name="data" type="CHAR_STRING" optional="true"/>
+      <arg name="Status" type="ContentLaunchStatusEnum"/>
+      <arg name="Data" type="CHAR_STRING" optional="true"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:48]: octet string here too

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml [hunk 3]
```diff
@@ -30,83 +30,84 @@ limitations under the License.
...
 
-  <struct name="Dimension">
+  <struct name="DimensionStruct">
     <cluster code="0x050a"/>
-    <item name="width" type="double" optional="false"/>
-    <item name="height" type="double" optional="false"/>
-    <item name="metric" type="MetricTypeEnum"/>
+    <item name="Width" type="double" optional="false"/>
+    <item name="Height" type="double" optional="false"/>
+    <item name="Metric" type="MetricTypeEnum"/>
```
Comment (andy31415) [inline src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:74]: should we set optional=false here as well? or is that the default for structs?

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:35]: Hmm.  That will break on-the-wire interop.  Let me think about this a bit.

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:35]: Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6204

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:48]: https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6204

File: src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml (hunk 3 above)
Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/content-launch-cluster.xml:74]: That's the default.

**Comment by bzbarsky-apple** [comment]
Note: I noticed that the LauncherResponse command was mis-named and fixed that, but now this will not pass CI until https://github.com/project-chip/zap/pull/901 is merged and we start using that ZAP version in Matter.

**Review by jtung-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**